cmd/imuad/external.go

```diff
@@ -0,0 +1,201 @@
+			mnemonic, _ := cmd.Flags().GetString(flagFeederMnemonic)
+			// TODO: refactor logger ?
+			logger := sdkserver.GetServerContextFromCmd(cmd).Logger.With("module", "price-feeder")
+			ok := pricefeeder.StartPriceFeeder(configFile, mnemonic, sourcesConfPath, logPath, logger)
+			if !ok {
+				return fmt.Errorf("failed to start price feeder")
+			}
+			return nil
+		},
+	}
+	feederCmd.Flags().String(flagConfFile, "", "file of feeder config")
+	feederCmd.Flags().String(flagSourcesConfPath, "", "path to sources config")
+	feederCmd.Flags().String(flagFeederLogPath, "", "path to feeder logs")
+	feederCmd.Flags().String(flagFeederMnemonic, "", "Oracle mnemonic")
+
+	_ = feederCmd.MarkFlagRequired(flagConfFile)
+	_ = feederCmd.MarkFlagRequired(flagSourcesConfPath)
+	return feederCmd
+}
+
+func feederStopCommand() *cobra.Command {
+	feederStopCmd := &cobra.Command{
+		Use:   "feeder-stop",
+		Short: "Stop the feeder subprocess started by --oracle",
+		RunE: func(_ *cobra.Command, _ []string) error {
+			pidData, err := os.ReadFile(feederPIDFile)
+			if err != nil {
+				return fmt.Errorf("failed to read feeder PID file: %w", err)
+			}
+
+			pidStr := strings.TrimSpace(string(pidData))
+			pid, err := strconv.Atoi(pidStr)
+			if err != nil {
+				return fmt.Errorf("invalid PID in file: %s", pidStr)
+			}
+
+			proc, err := os.FindProcess(pid)
+			if err != nil {
+				return fmt.Errorf("failed to find process: %w", err)
+			}
+
+			fmt.Printf("Sending SIGTERM to feeder (PID %d)...\n", pid)
+			if err := proc.Signal(syscall.SIGTERM); err != nil {
+				return fmt.Errorf("failed to send SIGTERM: %w", err)
+			}
+
+			_ = os.Remove(feederPIDFile)
+			fmt.Println("Feeder stopped successfully.")
+			return nil
+		},
+	}
+	return feederStopCmd
+}
+
+func launchFeeder(configFile, sourcesConfPath, binPath, mnemonic string, logger log.Logger, logPath string) {
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				logger.Error("feeder panic recovered", "err", r)
+			}
+		}()
+
+		if binPath != "" {
+			logger.Info("starting external feeder binary", "path", binPath)
+			startExternalFeeder(binPath, configFile, sourcesConfPath, logger, logPath)
+		} else {
+			logger.Info("starting feeder subprocess via CLI command")
+
+			selfPath, err := os.Executable()
+			if err != nil {
+				logger.Error("cannot determine self binary path", "err", err)
+				return
+			}
+
+			cmd := exec.Command(selfPath,
+				"external",
+				"feeder",
+				fmt.Sprintf("--%s", flagConfFile), configFile,
+				fmt.Sprintf("--%s", flagSourcesConfPath), sourcesConfPath,
+				fmt.Sprintf("--%s", flagFeederLogPath), logPath,
+				fmt.Sprintf("--%s", flagFeederMnemonic), mnemonic,
+			)
+
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+
+			if err := cmd.Start(); err != nil {
+				logger.Error("failed to start feeder subprocess", "err", err)
+				return
+			}
+
+			logger.Info("feeder subprocess started", "pid", cmd.Process.Pid)
+
+			_ = os.WriteFile(
```

Comment:
1. The current write operation is not atomic.
2. There is no prevention of multiple processes accessing the same PID file at the same time.